packages/dx-react-grid/src/plugins/sorting-state.jsx

```diff
@@ -32,11 +28,7 @@ export class SortingState extends React.PureComponent {
     );
 
     this.changeColumnSorting = stateHelper.applyReducer
-      .bind(stateHelper, (prevState, payload) => {
-        const keepOther =
-          calculateKeepOther(prevState.sorting, payload.keepOther, persistentSortedColumns);
-        return changeColumnSorting(prevState, { ...payload, keepOther });
-      });
+      .bind(stateHelper, getChangeColumnSorting(changeColumnSorting, props.columnExtensions));
```

Comment:
But the `columnExtensions` property can be changed dynamically. We can't add such a closure to the initial value